libp2p/utils.py

```diff
@@ -0,0 +1,95 @@
+import asyncio
+import struct
+from typing import Tuple
+
+from libp2p.typing import StreamReader
+
+TIMEOUT = 10
+
+
+def encode_uvarint(number: int) -> bytes:
+    """Pack `number` into varint bytes"""
+    buf = b""
+    while True:
+        towrite = number & 0x7F
+        number >>= 7
+        if number:
+            buf += bytes((towrite | 0x80,))
+        else:
+            buf += bytes((towrite,))
+            break
+    return buf
+
+
+def decode_uvarint(buff: bytes, index: int) -> Tuple[int, int]:
```

Comment:
are these encoding/decoding `LEB128` encoded integers?

Reply:
Yes, it is the unsigned LEB128. In go-libp2p, they use the function in standard library `binary.PutUvarint` to encode.

Reply:
maybe this other implementation I did in py-wasm might be a useful reference.  :+1: 

https://github.com/ethereum/py-wasm/blob/master/wasm/parsers/leb128.py